src/main/java/org/openstreetmap/atlas/checks/validation/tag/TunnelBridgeHeightLimitCheck.java

```diff
@@ -0,0 +1,154 @@
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        // case 1 (tunnel) & 2 (covered highway)
+        if ((TunnelTag.isTunnel(object) || this.isCovered(object))
+                && this.isHighwayWithoutMaxHeight(object))
+        {
+            final Long osmId = object.getOsmIdentifier();
+            markAsFlagged(osmId);
+            final int instructionIndex = TunnelTag.isTunnel(object) ? TUNNEL_CASE_INDEX
+                    : COVERED_CASE_INDEX;
+            return Optional
+                    .of(createFlag(object, getLocalizedInstruction(instructionIndex, osmId)));
+        }
+        // case 3 (road passing under bridge)
+        if (BridgeTag.isBridge(object))
+        {
+            final Edge bridge = (Edge) object;
+            final PolyLine bridgeAsPolyLine = bridge.asPolyLine();
+            final Set<Edge> edgesToFlag = new HashSet<>();
+            Iterables.stream(bridge.getAtlas().edgesIntersecting(bridge.bounds()))
+                    .filter(edge -> edge.isMainEdge()
+                            && edge.getOsmIdentifier() != bridge.getOsmIdentifier()
+                            && !isFlagged(edge.getOsmIdentifier())
+                            && this.isHighwayWithoutMaxHeight(edge)
+                            && this.edgeCrossesBridge(edge.asPolyLine(), bridgeAsPolyLine))
+                    .forEach(edge ->
+                    {
+                        markAsFlagged(edge.getOsmIdentifier());
+                        edgesToFlag.add(edge);
+                    });
+            if (!edgesToFlag.isEmpty())
+            {
+                final CheckFlag checkFlag = new CheckFlag(getTaskIdentifier(bridge));
+                edgesToFlag.forEach(
+                        edge -> checkFlag.addObject(edge, getLocalizedInstruction(BRIDGE_CASE_INDEX,
+                                edge.getOsmIdentifier(), bridge.getOsmIdentifier())));
+                return Optional.of(checkFlag);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    // check if the two polylines intersect at any location other than the bridge's endpoints
+    private boolean edgeCrossesBridge(final PolyLine edge, final PolyLine bridge)
+    {
+        return edge.intersections(bridge).stream()
+                .anyMatch(loc -> !loc.equals(bridge.first()) && !loc.equals(bridge.last()));
+    }
+
+    private boolean isCovered(final AtlasObject object)
+    {
+        return Validators.isOfType(object, CoveredTag.class, CoveredTag.YES, CoveredTag.ARCADE,
+                CoveredTag.COLONNADE);
+    }
+
+    private boolean isHighwayWithoutMaxHeight(final AtlasObject object)
+    {
+        return Validators.isOfType(object, HighwayTag.class, HighwayTag.MOTORWAY_LINK,
```

Comment:
We could use a highway tag configurable value here in case we'd ever want to change the list of acceptable highway tags.

Reply:
Thanks, Daniel. I have made the filter configurable.